larpandoracontent/LArObjects/LArMCParticle.h

```diff
@@ -137,15 +174,57 @@ inline pandora::StatusCode LArMCParticleFactory::Read(Parameters &parameters, pa
     // ATTN: To receive this call-back must have already set file reader mc particle factory to this factory
     int nuanceCode(0);
 
+    unsigned int nMCStepPositions(0);
+    std::vector<pandora::InputCartesianVector> mcStepPositions;
+
+    unsigned int nMCStepMomentas(0);
+    std::vector<pandora::InputCartesianVector> mcStepMomentas;
+
     if (pandora::BINARY == fileReader.GetFileType())
     {
         pandora::BinaryFileReader &binaryFileReader(dynamic_cast<pandora::BinaryFileReader&>(fileReader));
         PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, binaryFileReader.ReadVariable(nuanceCode));
+
+        PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, binaryFileReader.ReadVariable(nMCStepPositions));
+        for (int step = 0; step < nMCStepPositions; ++step)
+        {
+            pandora::CartesianVector mcStepPosition(0.0, 0.0, 0.0);
```

Comment:
Usually use `0.f` to make explicit that it's a single floating point literal that we want to assign to each element. Here and below.